osbenchmark/worker_coordinator/runner.py

```diff
@@ -698,6 +698,7 @@ async def __call__(self, opensearch, params):
                 tasks = await opensearch.tasks.list(params={"actions": "indices:admin/forcemerge"})
                 if len(tasks["nodes"]) == 0:
                     # empty nodes response indicates no tasks
+                    request_context_holder.on_client_request_end()
```

Comment:
This should be called just prior to the preceding `pass` statement, else there will be a mismatch between service time and the client processing time when an timeout exception is caught.

Reply:
Hey @gkamat, I've updated some code in my previous PR to ensure that request start is always calculated after the client request start, and that request end is always before the client request end. I'm sorry if I didn't address your concern properly. I believe that to get accurate metrics, request_context_holder.on_client_request_end() should be added after the task is completed. What do you think, @IanHoang and @VijayanB? Would love to hear your thoughts.

Reply:
@VijayanB, the time for the force merge reported is going to be inaccurate in any case, since the resolution with be the poll period, which is 10 sec.  It is just the degree of error, which should be weighed against the following.

The issue is how the client processing time is computed for multi-part operations.   In this case there is a truncated operation caused by the exception.  That should be handled on its own and the corresponding service time and associated client overhead computed.

Since there is a subsequent loop, both of these variables should be accumulators. Each time through the loop the poll period should be added to the service time.  Each call to the `list` operation will be a contributor as well, furthermore the client overhead of that call will be another metric that needs to be updated.

The root cause of this is how multi-part requests are handled, as was indicated to @saimedhi earlier.  Otherwise the concept of what the client overhead is for a particular call becomes ambiguous.

Reply:
Since this is holding the release back and the submitted fix will resolve the immediate problem, we can merge this in for now and address the larger issue subsequently.

Reply:
@VijayanB yes, polling is not part of force merge and the issue being referred to is not so much the inaccuracy in the duration reported as the accuracy of client overhead computation.  This is somewhat similar to, say, how scroll queries are handled, where the same issues are at play.